docs/design/interpreter/stackwalk.md

```diff
@@ -0,0 +1,25 @@
+# Stack walking for GC, exception handling and debugger
+To enable seamless integration of interpreter with GC, exception handling and debugger, we need to be able to walk stack that contains interleaved sequences of interpreted and AOT / JIT compiled frames. Fortunately, the same StackFrameIterator is used by the GC, exception handling and debugger, so adding a support to walk stack containing interpreter frames to this StackFrameIterator will give us support for all of the three scenarios.
+
+Here is the list of necessary changes:
+* Add a new explicit frame, the InterpreterFrame. Instance of this frame would be created in the mono_interp_exec_method. It will enable stack frame iterator to move through the interpreted frames managed by that mono_interp_exec_method. Please note that interpreter calls to other interpreted code are not recursively calling mono_interp_exec_method and each interpreted frame is represented by an instance of InterpFrame (these form a linked list).
```

Comment:
Should we make this sort of thing clear by having names like ClrInterpreterFrame and MonoInterpreterFrame?

Reply:
I guess that would make it even more confusing by naming existing Mono InterpFrame differently in the doc. I think we will end up renaming many data types in the interpreter, but if we rename it in this doc, the link between the doc and the real Mono state will be lost. 
I think I'll just rename the InterpreterFrame here to something different to make the doc less ambiguous and keep the InterpFrame.